modules/ROOT/pages/edits/sockets-connector.adoc

```diff
@@ -0,0 +1,94 @@
+
+== POM File Information
+
+[source,xml,linenums]
+----
+<dependency>
+  <groupId>org.mule.connectors</groupId>
+  <artifactId>mule-sockets-connector</artifactId>
+  <version>RELEASE</version>
+  <classifier>mule-plugin</classifier>
+</dependency>
+----
+
+Mule converts RELEASE to the latest version. To specify a version, view
+Anypoint Exchange and click *Dependency Snippets*.
+
+== Configure a Socket Listener
+
+The Socket Listener represents a socket endpoint that clients can connect to. By browsing to this endpoint, external clients establish a TCP Socket connection and trigger the flow each time they send a message.
+
+You can configure a Socket Listener by referencing the configuration element from the listener's `config-ref`. This configuration sets up a Socket Listener to accept local incoming connections on port 8082:
+
+[source,xml,linenums]
+----
+	<sockets:listener-config name="SocketsListenerConfig">
+		<sockets:tcp-listener-connection host="localhost" port="8082" />
+	</sockets:listener-config>
+----
+
+
+
+== Supported Operations
+
+Socket connector provides the user with two basic operations:
+
+    * Send
+    * Send and Receive
+
+== Supported Protocols
+
+Sockets Connector supports these TCP protocols:
+
+* Direct Protocol +
+  The socket reads until no more bytes are (momentarily) available. On slow networks EOFProtocol and LengthProtocol may be more reliable.
+* EOF Protocol +
+  Reading is terminated by the stream being closed by the client.
+* Length Protocol +
+  This protocol is defined by sending / reading an integer (the packet length) and then the data to be transferred.
+* Custom Class Loading Length Protocol +
+  A Length Protocol that uses a specific class loader to load objects from streams.
+* Safe Protocol +
+  This is a test protocol and should not be used in productive environments. This protocol precedes every message with a cookie.
+* Streaming Protocol +
+  This protocol allows the socket's send operation to return a message with the original InputStream as payload.
+* XML Message Protocol +
+  The XmlMessageProtocol is a protocol used to read streaming XML documents. The only requirement is that each document includes an XML declaration at the beginning of the document of the form "<?xml...". Make sure that the XML documents being streamed begin with an XML declaration when using this protocol. Data are read until a new document is found or until there is no more data (momentarily). For slower networks, XmlMessageEOFProtocol may be more reliable. Also, the default character encoding for the platform is used to decode the message bytes when looking for the XML declaration. Some caution with message character encodings is warranted.
+* XML Message Eof Protocol +
+  Extends XmlMessageProtocol to continue reading until either a new message or EOF is found.
+* Custom Protocol +
+  Users can define their own protocol by writing a class that extends TcpProtocol, and that will be used as custom protocol.
```

Comment:
"Users" or "You"?

Reply:
`TcpProtocol`?
Again, I think we need to distinguish between what we call a protocol and the actual syntax for the protocol as used in code. I have just guessed here; you will know best.